### ISSUE TITLE
Enclosing class/method info is confusing in declaration statements

### ISSUE THREAD

**Issue by nimakarimipour** [uber/NullAway#598]
`NullAway version: 0.9.6`

Assume that we have an interface called `Foo`:

```java
public interface Foo {
    void bar(Object o);
}
```
Now, assume an anonymous class is defined in a method called run like below:
```java
public class Main {
   public void run(){
      Foo foo = new Foo() {
         @Override
         public void bar(Object o) { }
      };
   }
}
```
If we make parameter o `@nullable` in `Foo.bar`. We will see an error reported in `Main$1` but the enclosing method is `run()` .
This makes two problems:
1. There is no method `run()` in `Main$1`.
2. If we want to investigate effectiveness of `@Nullable` on `Foo.bar.o`, since it is a method parameter the active regions are the method body and all sub-methods declarations,  in this case is {class: `Main$1` , method: `bar(Object)`} but the error is reported on {class: `Main$1` , method: `run()` }

The problem is `run()` is indeed the enclosing method where this error is reported, therefore the enclosing method term might be confusing. 

Possible solution is renaming `EnclosingClassAndMethodInfo` to `RegionInfo` and for all inheritance violation errors, target method itself should be the `regionInfo.method` itself.
